### ISSUE TITLE
(Optional?) randomized ports for isolated services

### ISSUE THREAD

**Issue by slint** [inveniosoftware/docker-services-cli#9]
## Is your feature request related to a problem? Please describe.

Imagine the following scenario:

- I'm running my separate Docker services for some other instance/application (e.g. Zenodo local development). These services are exposing standard ports, i.e. ES on 9200, PostgreSQL on 5432, etc.  
- I take a break to install a separate Invenio module, e.g. `invenio-records-rest` to make a bugfix.
- I run `invenio-records-rest`'s tests, but guess what `docker-services-cli` will have to either:
  - Fail hard and exit, since it will try to bring up services that have already bound ports (the ones from my Zenodo local dev Docker services). This is actually a good/preferred behavior 
  - Partially/silently fail and re-use my existing Docker services. This is pretty bad, since as part of tests it might end up modifying/wiping the services, which affect my Zenodo local development setup.

## Describe the solution you'd like

I think checking if the current implementation of `docker-services-cli` fails hard (and possibly with an informative message) in the above use-case is the bare minimum to make sure things are safe.

An alternative, more "sophisticated" solution, would be to bring up each Docker service with a randomized available port (preferably in a higher port range, e.g. 50000+), that can then be exposed to the underlying tests, e.g. via environment variables. A downside of the randomized ports solution is that if one wants mid-test to inspect one of the Docker services state, they have to look up what is the current randomized port and modify their connection parameters accordingly.

Another alternative is to standardized these "testing" ports, so that we avoid clashes with other local development ports. E.g. we put ES always on 59200, PostgreSQL always on 55432, etc.

To summarize here are the available solutions:

- [ ] Check that we currently fail hard...
  - [ ] ...with an informative message
- [ ] Randomized ports, exposed in the output as environment variable
- [ ] Decide and use alternative "standardized" ports

**Comment by slint** [comment]
Closed via #39
